geo/geomfn: fix missing free in GeneratePoints

otan opened the pull request:
Looks like I missed this in an earlier review -- we do not free a
PreparedGeometry that we created.

I've also changed the Destroy not returning to panic, as I feel as if we
are not freeing correctly something is deeply wrong in the system which
can lead to other scary errors such as OOM.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/60280)
<!-- Reviewable:end -->

otan committed:
geo/geomfn: fix missing free in GeneratePoints

Looks like I missed this in an earlier review -- we do not free a
PreparedGeometry that we created.

I've also changed the Destroy not returning to panic, as I feel as if we
are not freeing correctly something is deeply wrong in the system which
can lead to other scary errors such as OOM.

Release note: None

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/60280#-:-MTBcSNNAv0P2895hTNg:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
thanks

bors r=sumeerbhola
